cogs/profiles.py

```diff
@@ -178,50 +177,47 @@ async def profile(self, ctx, *, userName:str = None):
         profiles = data_handler.load("profiles")
         userids = list()
 
-        # if user wants to display his own profile, display only his own.
         if userName is None:
-            foundUserName = ctx.author.name
+            # if user wants to display his own profile, display only his own.
+            userid = ctx.message.author.id
         else:
-            foundUserName = userName
+            for user in list(filter(lambda u: userName in u.name, self.bot.users)):
+                userids.append(user.id)
 
-        users = list(filter(lambda u: foundUserName in u.name, self.bot.users))
-        for user in users:
-            userids.append(user.id)
-
-        if userName is not None:
             for guild in self.bot.guilds:
-                members = list(filter(lambda m: foundUserName in m.display_name, guild.members))
-                for member in members:
+                for member in list(filter(lambda m: userName in m.display_name, guild.members)):
                     userids.append(member.id)
 
             for profil in profiles:
-                if foundUserName in profiles[profil]['Base']['username']:
+                if userName in profiles[profil]['Base']['username']:
                   userids.append(int(profil))
```

Comment:
Can `profil` be renamed to `profile` so that it's no longer a typo?

Reply:
Fixed :)